// mn_api_tuling/postToRobot.js
/**
 * Created by Administrator on 2018/5/17.
 */

'use strict';
var pageName = 'postToRobot';


//var com = require ('mn_fun_comm');

var post = require ('./post_http');
/**
 *
 * @param data{string}
 * @param callback
 */
function postData (data,callback) {
    var host = 'www.tuling123.com';
    var path = '/openapi/api/v2';
    post(host,path,data,callback);

};


var postToRobot =function(content,options,callback){

    var host = 'www.tuling123.com';
    var path = '/openapi/api/v2';

    var userInfo = options;
    var txtInput = content ;

    var data = {
        "reqType":0,
        "perception": {
            "inputText": { "text": txtInput  }
        },
        "userInfo": userInfo
    };

    var content = JSON.stringify(data);
    post (host,path,content,function(err,dataCallback){
        if (err) {
            console.log (pageName + ':there is an error:' + err);
        }
        var dataCallbackJson =JSON.parse(dataCallback);
        console.log (pageName + ":dataCallback:" + dataCallback);
        console.log (pageName + ":dataCallback:results:" + JSON.stringify (dataCallbackJson.results));
        //console.log (pageName + ":dataCallback:values:" + JSON.stringify (dataCallbackJson.results[0].values));
        var code = dataCallbackJson.intent.code;
        if (code >= 10000) {
            var result = dataCallbackJson.results[0];

            var resultType = dataCallbackJson.results[0].resultType;
            var callbackFromRobot={};
            switch (resultType) {
                case 'text':
                    var textCallback=dataCallbackJson.results[0].values.text;
                    console.log (pageName + ':callback:' + textCallback);

                    callbackFromRobot = {
                        resultType : 'text',
                        content : textCallback
                    };
                    callback(null,callbackFromRobot);
                    break;
                case 'image':

                    callbackFromRobot = {
                        resultType : 'image',
                        imageUrl : result.values.image
                    };

                    break;
                case  'voice':

                    break;
                case 'url' :
                    resultType :'url';
                    url : result.vaules.url

                    break;
                case  'video':

                    break;
                case 'news':

                    break;


            }


        }


        /**
         *
         *  callback a image if resultType in results of callback data is image .
         * :[{"groupType":0,
         * "resultType":"image",
         * "values":{"silentState":"0f9b66e1-a6b8-43d1-ab6e-639e6d594ecc",
         * "image":"http://turing-chat.oss.tuling123.com/d2b88fc28970dfbdc4b692a0f5bc2e07.jpg"}}]
         *
         *
         *  results:[{"groupType":1,"resultType":"url","values":{"url":"http://m.image.so.com/i?q=%E5%9B%9E"}},
         *  {"groupType":1,"resultType":"text","values":{"text":"亲，已帮你找到图片"}}]

         *
         *
         */


    } );

};



var apiTuling = function(content,callback){
    postData(content,callback);
    //postToRobot(content,options, callback);

};

module.exports = apiTuling;
